src/main/scala/chisel3/testers/BasicTester.scala

```diff
@@ -9,6 +9,7 @@ import chisel3.internal.Builder.pushCommand
 import chisel3.internal.firrtl._
 import chisel3.internal.sourceinfo.SourceInfo
 
+@deprecated("This is not a User API, will be removed in Chisel 3.6", "3.5")
 class BasicTester extends Module() {
   // The testbench has no IOs, rather it should communicate using printf, assert, and stop.
   val io = IO(new Bundle() {})
```

Comment:
My main concern with deprecating this is it is the only API exposing `stop()` (which is I guess the real problem). A user cannot replicate the behavior of this unless we add `stop`. 

Also we should point people to chiseltest rather than just saying "This is not a User API"

Reply:
> A user cannot replicate the behavior of this unless we add stop.

`chisel3.stop` was a user API, originally:
```
  def stop()(implicit sourceInfo: SourceInfo) {
    // TODO: rewrite this using library-style SourceInfo passing.
    when (!reset.asBool) {
      pushCommand(Stop(sourceInfo, clock.ref, 0))
    }
  }
```
I think this is `chisel3.stop` too?

> Also we should point people to chiseltest rather than just saying "This is not a User API"

Yes, I should change the deprecation wording.